packages/nectar/src/sagas/publicChannels/checkForMessages/checkForMessages.saga.test.ts

```diff
@@ -1,122 +1,90 @@
+import { DateTime } from 'luxon'
 
 describe('checkForMessagesSaga', () => {
-  const community: Community = {
-    name: '',
-    id: 'id',
-    CA: null,
-    rootCa: '',
-    peerList: [],
-    registrarUrl: 'registrarUrl',
-    registrar: null,
-    onionAddress: '',
-    privateKey: '',
-    port: 0
-  }
+  let store: Store
+  let factory: FactoryGirl
 
-  const messages = [
-    {
-      id: '1',
-      type: 0,
-      message: 'message',
-      createdAt: 0,
-      channelId: '',
-      signature: '',
-      pubKey: ''
-    }
-  ]
+  let community: Community
+  let alice: Identity
+  let channel: PublicChannel
 
-  const communityChannels: CommunityChannels = {
-    id: 'id',
-    currentChannel:
-      'zs10zkaj29rcev9qd5xeuzck4ly5q64kzf6m6h9nfajwcvm8m2vnjmvtqgr0mzfjywswwkwke68t00',
-    channels: publicChannelsAdapter.getInitialState(),
-    channelMessages: channelMessagesAdapter.setAll(
-      channelMessagesAdapter.getInitialState(),
-      messages
-    ),
-    channelLoadingSlice: 0
-  }
+  beforeAll(async () => {
+    setupCrypto()
 
-  const identity: Identity = {
-    id: 'id',
-    hiddenService: {
-      onionAddress: 'onionAddress.onion',
-      privateKey: 'privateKey'
-    },
-    dmKeys: { publicKey: 'publicKey', privateKey: 'privateKey' },
-    peerId: { id: 'peerId', pubKey: 'pubKey', privKey: 'privKey' },
-    zbayNickname: '',
-    userCsr: undefined,
-    userCertificate: ''
-  }
+    // Set date display format
+    process.env.LC_ALL = 'en_US.UTF-8'
 
-  test.skip('ask for missing messages', async () => {
-    await expectSaga(checkForMessagesSaga)
-      .withReducer(
-        combineReducers({
-          [StoreKeys.PublicChannels]: publicChannelsReducer,
-          [StoreKeys.Communities]: communitiesReducer,
-          [StoreKeys.Identity]: identityReducer
-        }),
-        {
-          [StoreKeys.PublicChannels]: {
-            ...new PublicChannelsState(),
-            channels: communityChannelsAdapter.setAll(
-              communityChannelsAdapter.getInitialState(),
-              [communityChannels]
-            )
-          },
-          [StoreKeys.Communities]: {
-            ...new CommunitiesState(),
-            currentCommunity: 'id',
-            communities: communitiesAdapter.setAll(
-              communitiesAdapter.getInitialState(),
-              [community]
-            )
+    store = prepareStore().store
+
+    factory = await getFactory(store)
+
+    community = await factory.create<
+    ReturnType<typeof communitiesActions.addNewCommunity>['payload']
+    >('Community')
+
+    channel = publicChannels.selectors.publicChannels(store.getState())[0]
+
+    alice = await factory.create<ReturnType<typeof identityActions.addNewIdentity>['payload']>(
+      'Identity',
+      { id: community.id, zbayNickname: 'alice' }
+    )
+  })
+
+  test('ask for missing messages', async () => {
+    const types = [MessageType.Basic, MessageType.Empty, MessageType.Empty]
+
+    const shuffled = types
```

Comment:
Curious: why are we shuffling?

Reply:
It probably not necessary but I wanted to make sure the mechanism will never depend on the messages order